macos-pkg-builder/build-macos.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+
+SCRIPTPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+TARGET_DIRECTORY="$SCRIPTPATH/target"
+PRODUCT="OpenSearchCLI"
+DATE=`date +%Y-%m-%d`
+TIME=`date +%H:%M:%S`
+LOG_PREFIX="[$DATE $TIME]"
+
+function usage() {
+    echo ""
+    echo "This script is used to build the macos package for OpenSearch CLI."
+    echo "--------------------------------------------------------------------------"
+    echo "Usage: $0 [args]"
+    echo ""
+    echo "Required arguments:"
+    echo -e "-v VERSION\tSpecify the OpenSearch CLI version number that you are building, e.g. '1.0.0' or '1.0.0-beta1'."
+    echo -e "-a ARCHITECTURE\tSpecify the architecture type your signed binary is based on, e.g. 'x64,arm64'."
+    echo "--------------------------------------------------------------------------"
+}
+
+while getopts ":hv:a:" arg; do
+    case $arg in
+        h)
+            usage
+            exit 1
+            ;;
+        v)
+            VERSION=$OPTARG
+            ;;
+        a)
+            ARCHITECTURE=$OPTARG
+            ;;
+        :)
+            echo "-${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+    esac
+done
+
+# Validate the required parameters to present
+if [ -z "$VERSION" ] || [ -z "$ARCHITECTURE" ]; then
+  echo "You must specify '-v VERSION', '-a ARCHITECTURE'"
+  usage
+  exit 1
+fi
+if [ "$ARCHITECTURE" != "x64" ] && [ "$ARCHITECTURE" != "arm64" ]; then
+	echo "We only support 'x64' and 'arm64' as architecture name for -a parameter"
+  exit 1
+fi
+
+INSTALLER_NAME="opensearch-cli-${VERSION}-macos-${ARCHITECTURE}"
+#Functions
+go_to_dir() {
+    pushd $1 >/dev/null 2>&1
+}
+
+log_info() {
+    echo "${LOG_PREFIX}[INFO]" $1
+}
+
+log_error() {
+    echo "${LOG_PREFIX}[ERROR]" $1
+}
+
+deleteInstallationDirectory() {
+    log_info "Cleaning $TARGET_DIRECTORY directory."
+    rm -rf $TARGET_DIRECTORY
```

Comment:
Why not make target dir an argument, as is done in previous functions?

Reply:
Ack. Added as argument.